### PR TITLE
Improve docs: Getting started

### PR DESCRIPTION
**Pull request by jordisala1991**
I am targeting this branch, because this is docs.

Closes #4952 
Closes #4951 

## Subject

This includes improvements for the getting started documentation

Based on those two PR I created a new one:
- Adapt instructions for Symfony 4, with a note at the top to inform that if you are not using Symfony 4 there are some thing that might need to change
- Minor improvements on images and newlines
- Added configuration-block to yaml code 
- Change string form types to FQCNs that is compatible with Symfony 2.8, 3 and 4

### CODE REVIEW

File: docs/getting_started/creating_an_admin.rst [hunk 1]
```diff
@@ -5,6 +5,10 @@ You've been able to get the admin interface working in :doc:`the previous
 chapter <installation>`. In this tutorial, you'll learn how to tell SonataAdmin
 how an admin can manage your models.
 
+.. note::
+    This article assumes you are using Symfony 4. Using Symfony 2.8 or 3 will require
```
Comment (kunicmarko20) [inline docs/getting_started/creating_an_admin.rst]: Please comply with the sf docs standards:

> [Each line should break approximately after the first word that crosses the 72nd character (so most lines end up being 72-78 characters);](http://symfony.com/doc/current/contributing/documentation/standards.html)

Comment (jordisala1991) [reply docs/getting_started/creating_an_admin.rst]: Done

File: docs/getting_started/creating_an_admin.rst [hunk 2]
```diff
@@ -14,14 +19,14 @@ using these commands:
...
 .. code-block:: bash
 
-    $ php bin/console doctrine:generate:entity --entity="AppBundle:Category" --fields="name:string(255)" --no-interaction
-    $ php bin/console doctrine:generate:entity --entity="AppBundle:BlogPost" --fields="title:string(255) body:text draft:boolean" --no-interaction
+    $ php bin/console doctrine:generate:entity --entity="App:Category" --fields="name:string(255)" --no-interaction
+    $ php bin/console doctrine:generate:entity --entity="App:BlogPost" --fields="title:string(255) body:text draft:boolean" --no-interaction
 
 After this, you'll need to tweak the entities a bit:
 
 .. code-block:: php
```
Comment (kunicmarko20) [inline docs/getting_started/creating_an_admin.rst]: Please comply with the sf docs standards:

> [The :: shorthand is preferred over .. code-block:: php to begin a PHP code block;](http://symfony.com/doc/current/contributing/documentation/standards.html)

File: docs/getting_started/creating_an_admin.rst [hunk 3]
```diff
@@ -50,7 +55,7 @@ Set the default value to ``false``.
...
```
Comment (kunicmarko20) [inline docs/getting_started/creating_an_admin.rst]: same here

File: docs/getting_started/creating_an_admin.rst [hunk 4]
```diff
@@ -69,8 +74,7 @@ Set the default value to ``false``.
...
```
Comment (kunicmarko20) [inline docs/getting_started/creating_an_admin.rst]: same here

File: docs/getting_started/creating_an_admin.rst [hunk 5]
```diff
@@ -126,19 +129,20 @@ easiest way to do this is by extending ``Sonata\AdminBundle\Admin\AbstractAdmin`
...
```
Comment (kunicmarko20) [inline docs/getting_started/creating_an_admin.rst]: same here

File: docs/getting_started/creating_an_admin.rst [hunk 6]
```diff
@@ -174,18 +178,20 @@ SonataAdminBundle to know that this Admin class exists. To tell the
 SonataAdminBundle of the existence of this Admin class, you have to create a
```
Comment (kunicmarko20) [inline docs/getting_started/creating_an_admin.rst]: over 72

File: docs/getting_started/creating_an_admin.rst [hunk 7]
```diff
@@ -174,18 +178,20 @@ SonataAdminBundle to know that this Admin class exists. To tell the
...
 
-    services:
-        # ...
-        admin.category:
-            class: AppBundle\Admin\CategoryAdmin
-            arguments: [~, AppBundle\Entity\Category, ~]
-            tags:
-                - { name: sonata.admin, manager_type: orm, label: Category }
-            public: true
+        # app/config/services.yml
```
Comment (kunicmarko20) [inline docs/getting_started/creating_an_admin.rst]: shouldn't this be path for flex?

File: docs/getting_started/creating_an_admin.rst [hunk 8]
```diff
@@ -198,15 +204,17 @@ Step 4: Register SonataAdmin custom Routes
 SonataAdminBundle generates routes for the Admin classes on the fly. To load these
 routes, you have to make sure the routing loader of the SonataAdminBundle is executed:
 
-.. code-block:: yaml
+.. configuration-block::
+
+    .. code-block:: yaml
 
-    # app/config/routing.yml
+        # app/config/routing.yml
```
Comment (kunicmarko20) [inline docs/getting_started/creating_an_admin.rst]: same here?

File: docs/getting_started/the_form_view.rst [hunk 9]
```diff
@@ -7,15 +7,20 @@ discover! In the coming chapters, you'll create an Admin class for the more
...
+    This article assumes you are using Symfony 4. Using Symfony 2.8 or 3
+    will require to slightly modify some namespaces and paths when creating
+    entities and admins.
+
 Bootstrapping the Admin Class
 -----------------------------
 
 The basic class definition will look the same as the ``CategoryAdmin``:
 
 .. code-block:: php
```
Comment (kunicmarko20) [inline docs/getting_started/the_form_view.rst]: use `::`

File: docs/getting_started/creating_an_admin.rst (hunk 2 above)
Comment (jordisala1991) [reply docs/getting_started/creating_an_admin.rst]: Give me an example on sonata docs, to use it.

File: docs/getting_started/the_form_view.rst [hunk 10]
```diff
@@ -66,19 +71,22 @@ The ``BlogPost`` model has 4 properties: ``id``, ``title``, ``body``,
 database. This means the form view just needs 3 fields: title, body and
 category.
 
-The title and body fields are simple "text" and "textarea" fields, you can add
-them straight away:
+The title and body fields are simple ``TextType`` and ``TextareaType`` fields,
+you can add them straight away:
 
 .. code-block:: php
```
Comment (kunicmarko20) [inline docs/getting_started/the_form_view.rst]: `::`

File: docs/getting_started/creating_an_admin.rst (hunk 6 above)
Comment (jordisala1991) [reply docs/getting_started/creating_an_admin.rst]: I didn't touch that line

File: docs/getting_started/the_form_view.rst [hunk 11]
```diff
@@ -88,32 +96,35 @@ Adding Fields that Reference Other Models
 -----------------------------------------
 
 You have a couple different choices on how to add fields that reference other
-models. The most basic choice is to use the `entity field type`_ provided by
-the DoctrineBundle. This will render a choice field with the available entities
+models. The most basic choice is to use the ``EntityType`` provided by
+the Doctrine Bridge. This will render a choice field with the available entities
```
Comment (kunicmarko20) [inline docs/getting_started/the_form_view.rst]: over 72.

File: docs/getting_started/creating_an_admin.rst (hunk 7 above)
Comment (jordisala1991) [reply docs/getting_started/creating_an_admin.rst]: Isn't this path correct on flex?

File: docs/getting_started/the_form_view.rst [hunk 12]
```diff
@@ -88,32 +96,35 @@ Adding Fields that Reference Other Models
...
 as choice.
 
 .. code-block:: php
```
Comment (kunicmarko20) [inline docs/getting_started/the_form_view.rst]: `::`

File: docs/getting_started/the_form_view.rst [hunk 13]
```diff
@@ -122,27 +133,33 @@ Using the Sonata Model Type
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 To make life easier for admins, you can use the
-:ref:`sonata_type_model field type <field-types-model>`. This field type will
+:ref:`ModelType field <field-types-model>`. This field type will
 also render as a choice field, but it includes a create button to open a
 dialog with the admin of the referenced model in it:
 
 .. code-block:: php
```
Comment (kunicmarko20) [inline docs/getting_started/the_form_view.rst]: `::`

File: docs/getting_started/the_form_view.rst [hunk 14]
```diff
@@ -156,20 +173,25 @@ category field to a Meta data group. To do this, use the ``with()`` method:
...
```
Comment (kunicmarko20) [inline docs/getting_started/the_form_view.rst]: `::`

File: docs/getting_started/the_form_view.rst [hunk 15]
```diff
@@ -182,7 +204,7 @@ order to tweak the styling:
...
```
Comment (kunicmarko20) [inline docs/getting_started/the_form_view.rst]: you get the point :smile:

File: docs/getting_started/creating_an_admin.rst (hunk 2 above)
Comment (kunicmarko20) [reply docs/getting_started/creating_an_admin.rst]: https://raw.githubusercontent.com/sonata-project/SonataAdminBundle/3.x/docs/cookbook/recipe_custom_view.rst you just replace `.. code-block:: php` with `::`, /cc @greg0ire we are doing this, right?

File: docs/getting_started/creating_an_admin.rst (hunk 7 above)
Comment (kunicmarko20) [reply docs/getting_started/creating_an_admin.rst]: no, it should be `config/services/yaml`

File: docs/getting_started/creating_an_admin.rst (hunk 2 above)
Comment (jordisala1991) [reply docs/getting_started/creating_an_admin.rst]: Im not gonna argue against symfony rules, if they follow them: https://raw.githubusercontent.com/symfony/symfony-docs/master/best_practices/controllers.rst

And I didn't even search for that, just the first one that I open

Comment (greg0ire) [reply docs/getting_started/creating_an_admin.rst]: It's preferred, but not mandatory I guess?

Comment (jordisala1991) [reply docs/getting_started/creating_an_admin.rst]: http://symfony.com/doc/current/contributing/documentation/standards.html#example

I don't even understand this example xD
Don't know, if you want me to change this, I change it, but it is not clear when you should use short notation

Comment (greg0ire) [reply docs/getting_started/creating_an_admin.rst]: See https://github.com/sonata-project/SonataAdminBundle/pull/4452

Comment (kunicmarko20) [reply docs/getting_started/creating_an_admin.rst]: Well if it is not mandatory then we can stick with `.. code-block:: php`?

Comment (greg0ire) [reply docs/getting_started/creating_an_admin.rst]: We can do both I guess, depends on how you interpret it `¯\_(ツ)_/¯`

**Comment by jordisala1991** [comment]
Please review again. I changed to short notation when it makes sense to me. When the code does not have any text before like the http://symfony.com/doc/current/contributing/documentation/standards.html#example example I didn't use the short notation.

**Comment by greg0ire** [comment]
I'm on my phone, can either of you open a PR on the sf docs to remove that shorthand rule? Might be the best way to clarify things.

**Comment by jordisala1991** [comment]
IMO I would prefer to get docs improved without touching that block-code notation. And if someone wants to know what is the prefered case, then change all docs at once.

Can I revert the last commit?

**Comment by greg0ire** [comment]
I think you can

**Comment by jordisala1991** [comment]
I removed it because I am not sure if that is a good approach. It might be good but the rest of the docs are not written that way, and it probably should be changed all at once after clarifying with Symfony

**Comment by greg0ire** [comment]
Thanks @jordisala1991 !

**Pull request merged**